src/test/ui/issues/issue-101998/issue-101998.stderr

```diff
@@ -0,0 +1,28 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-101998.rs:3:5
+   |
+LL | fn parse_num(num: String) -> i32 {
+   |                              --- expected `i32` because of return type
+LL |     result_i32_error() // specific Result error message
+   |     ^^^^^^^^^^^^^^^^^^
+   |     |
+   |     expected `i32`, found enum `Result`
+   |     help: Either change the return type or use `unwrap`: `result_i32_error().unwrap()`
```

Comment:
Instead of mentioning to change the return type here, you could create a second suggestion at the `--- expected `i32` because of return type` span that suggests the found type.

Reply:
That should also resolve your concern

> the current error message "change return type to" doesn't make sense in this context.

Reply:
Yeah that makes a lot of sense.  Do you know how I can get the `span` of that part?  I can get the span of `result_i32_error()` with `trace.cause.span()`, but where can I find the span of `-> i32`?  Is that in `trace.values`?

Reply:
Some archeology later (this is happening at a *caller* to this function instead of here 🤦 ) I am guessing that the note is being emitted here:

https://github.com/rust-lang/rust/blob/3c2d20ef0b9cbb6f7c2cbc8112bb2e9f0d8ef755/compiler/rustc_typeck/src/check/fn_ctxt/suggestions.rs#L703

So... The fix would probably be to extend `ExpectedReturnTypeLabel::Other` with an `actual` field and then figure out how to emit a suggestion wherever the `Other` variant is processed.